File: Labs/Lab5(2.2)/python/symbols.py
from abc_symbols import ABC_symbol
from line import Line

class Symbols(Line, ABC_symbol):
    def __init__(self, symbols):
        if type(symbols) == Symbols:
            self.__line = Symbols(symbols.get_symbols())
        elif type(symbols) == str: 
            self.__line:str = symbols
        else:
            self.__line = ''
    
    def get_len(self):
        return super().get_len(self.__line)
    
    def get_symbol_amount(self) -> str:
        return super().get_symbol_amount(self.__line, '*')
    
    def get_symbols(self):
        return self.__line
    
    def __str__(self):
        return f"Symbols: {self.__line}, Length: {self.get_len()}, Amount of '*': {self.get_symbol_amount()}"

File: Labs/Lab5(2.2)/c#/Symbols.cs
namespace LineApp
{
    public class Symbols : Line
    {
        private readonly string _line;

        public Symbols()
        {
            _line = string.Empty;
        }

        public Symbols(string line)
        {
            _line = line;
        }

        public Symbols(Symbols other) : base()
        {
            _line = other.GetSymbols();
        }

        public override int GetLen(string line)
        {
            int uppercaseCount = 0;

            foreach (char c in line)
            {
                if (!char.IsDigit(c))
                {
                    uppercaseCount++;
                }

            }
            return uppercaseCount;
        }

        public override int GetSymbolAmount(string line, char symbol = '*')
        {
            return base.GetSymbolAmount(_line, symbol);
        }

        public string GetSymbols()
        {
            return _line.ToUpper();
        }

        public override string ToString()
        {
            return $"Symbols: {_line}, Length: {GetLen(_line)}, Amount of '*': {GetSymbolAmount(_line)}";
        }
    }
}